CSharp/Library/Microsoft.Bot.Connector.Shared/HttpClientEx.cs

```diff
@@ -19,7 +19,7 @@ public static class HttpClientEx
         public static async Task AddAPIAuthorization(this HttpClient client, string appId = null, string password = null)
         {
             var token = await new MicrosoftAppCredentials(appId, password).GetTokenAsync();
-            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token.AccessToken);
```

Comment:
This is an unexpected change. Is the old code broken? Passing in token vs token.AccessToken seems like a major difference in behavior.

Reply:
Seems like just a change in the structures we pass around. Everything in the world talks access_token so there's no way that the old `token` value represents the whole OAuth reponse.

Reply:
Exactly. The old `token` was the AccessToken alone. Now that we use Adal, we get an object encapsulating the AccessToken plus a lot of handy information, such as expiration date.

Reply:
I felt like it was more valuable to return the entire object and have the callers use jsut the token or the entire thing.